source/SH3/system/glprogram.cpp

```diff
@@ -75,11 +80,17 @@ GLuint sh3_glprogram::Compile(std::uint32_t type)
 
         source.open(fname);
 
+        if(!source.is_open())
+            die("sh3_glprogram::Compile( ): Unable to open a handle to %s!", fname.c_str());
+
         while(!source.eof())
```

Comment:
Ahh I see. I'm using Windows, so I think the problem is stemming from the stupid `\r\n` at the end of each line. 

Why is this method safer?